app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
+
+    attr_reader :raw_value
+
+    def self.values
+      valid_values.map { |value| new(value) }
+    end
+
+    def self.valid_values
+      %w(hidden optional required).freeze
+    end
+
+    def initialize(raw_value)
+      @raw_value = raw_value
+      freeze
```

Comment:
What order does rails call these in? If we have both the validation on product and throw an exception within the UserNoteMode, will we get the validation error or the hard exception first? If the validation error happens first, is there any downside to having a fallback check within the UserNoteMode object?